File: backend/core-service/src/Fix.Application/UseCases/Orders/Dtos/OrderDto.cs
using Fix.Application.Mandates.Dtos;
using Fix.Domain.Common;
using Fix.Domain.AggregateRoots.Orders;

namespace Fix.Application.Orders.Dtos;

/// <summary>Boleta de hedge, com enquadramento e carimbos de desvio (sem mandato, estouro, a posteriori).</summary>
public sealed record OrderDto(
    Guid Id,
    string Code,
    Guid? MandateId,
    string? MandateCode,
    string MandateTitle,
    OrderType Type,
    TradeDirection Direction,
    Guid CounterpartyId,
    string CounterpartyName,
    Commodity? Commodity,
    string Tenor,
    decimal? Lots,
    decimal? NotionalUsd,
    decimal Price,
    string PriceUnit,
    OptionKind? OptionKind,
    decimal? Premium,
    bool CoveredSale,
    DateOnly TradeDate,
    string? Notes,
    ComplianceDto Compliance,
    bool LinkedAfterExecution,
    bool ExceedsMandate,
    string? DeviationNote,
    ApprovalStatus Approval,
    Guid RequestedBy,
    Guid? DecidedBy,
    DateTimeOffset? DecidedAt,
    string? DecisionNote,
    ConfirmationStatus Confirmation,
    DateOnly? ConfirmedOn,
    string? ConfirmationNote,
    Guid? ConfirmationBy,
    bool ConfirmationOverdue);

File: frontend/bff/src/application/orders/dtos/order.dto.ts
import type { ApprovalStatus } from '../../../cross-cutting/enums/approval-status.js';
import type { Commodity } from '../../../cross-cutting/enums/commodity.js';
import type { ConfirmationStatus } from '../../../cross-cutting/enums/confirmation-status.js';
import type { OrderTermsDto } from './order-terms.dto.js';

export interface OrderDto {
  id: string;
  mandateId: string;
  mandateTitle: string;
  counterpartyId: string;
  counterpartyName: string;
  /** Ausente em NDF (hedge de moeda, sem commodity). */
  commodity: Commodity | null;
  terms: OrderTermsDto;
  approval: ApprovalStatus;
  requestedBy: string;
  decidedBy: string | null;
  decidedAt: string | null;
  decisionNote: string | null;
  /** Nasce pendente; só pode ser tratado pelo middle office depois da aprovação. */
  confirmation: ConfirmationStatus;
  confirmedOn: string | null;
  confirmationNote: string | null;
  /** Pendente há mais de 2 dias úteis. */
  confirmationOverdue: boolean;
}
